File: API/src/params/facility.params.ts
import { PaginationParams } from "./pagination.params";

export class FacilityParams extends PaginationParams {
  minPrice: number = 0;
  maxPrice: number = 1000000;
  lat: number = 0;
  lon: number = 0;
  sortBy: string = "location";
  order: string = "asc";
  province?: string | undefined;
}

File: BadcourtAPI/SharedKernel/Params/FacilityParams.cs
namespace SharedKernel.Params;

public class FacilityParams : PaginationParams
{
    public string? UserId { get; set; }
    public string? FacilityName { get; set; }
    public double Lat { get; set; }
    public double Lon { get; set; }
    public string? Province { get; set; }
    public decimal MinPrice { get; set; } = decimal.MinValue;
    public decimal MaxPrice { get; set; } = decimal.MaxValue;
    public string? Search { get; set; }
    public string OrderBy { get; set; } = "registeredAt";
    public string SortBy { get; set; } = "desc";
}
